docs/UpgradeGuide.md

```diff
@@ -0,0 +1,568 @@
+# Upgrading ddtrace
+
+# From 0.x to 1.0
+
+- [Namespacing & the public API](#namespacing--the-public-api)
+- [Configuration](#configuration)
+- [Instrumentation](#instrumentation)
+- [Full list of breaking changes](#full-list-of-breaking-changes)
+
+Upgrading `ddtrace` from 0.x to 1.x introduces some changes to namespacing, the public API, and the underlying trace data structure.
+
+Here's a list of the most common changes you may encounter:
+
+## Namespacing & the public API
```

Comment:
I like how through this document is, BUT I do think it's looking kinda scary.

Since we already know that many customers don't upgrade often, I think we need a section that TL;DRs this document for customers that just turn on dd-trace-rb without using any of its APIs.

E.g. something like

> ## For all customers
>
> Review your configuration, change it as follows [...], and you're all done.
>
> ## For customers doing custom tracing and ....
>
> Here's all of the things that may matter to you. [...]

...that reassures most customers that actually moving from 0.x to 1.0 can be as simple as not changing a single line of code (if no config to change), or change only a handful of lines (adapt your config).

Otherwise this document makes it look like a quite involved task, and I suspect that may turn away people from going to 1.0, which would be a big problem.

Reply:
I agree in concept, it's just not clear to me how to execute this without duplicating a lot of information.

It may make sense to have a significantly truncated version of this in the 1.0 milestone release notes itself, then link to this guide as a "for more details or custom instrumentation see this guide" kinda thing.

Reply:
Ahhh, I didn't see the full list of all the namespace changes (which I agree is pretty significant.) Some of this more detailed information I think can be moved to a glossary/reference section at the bottom of the document, so as to keep the initial read light and breezy, but with access to richer info if needed.

Reply:
I moved all the comprehensive lists of changes into tables in an appendix section at the bottom of the document. Should help keep the focus on the meat of what changed, instead of overwhelming the reader with details.